File: laramie-importacion-webapi/src/domain/entities/recaudacion.ts
export default class Recaudacion {

	numeroControl: string;
	numeroComprobante: string;
	codigoTipoTributo: string;
	numeroCuenta: string;
	codigoDelegacion: string;
	numeroRecibo: number;
	importeCobro: number;
	fechaCobro: Date;
	codigoBarras: string;
	observacion: string;

	constructor(
		numeroControl: string = "",
		numeroComprobante: string = "",
		codigoTipoTributo: string = "",
		numeroCuenta: string = "",
		codigoDelegacion: string = "",
		numeroRecibo: number = 0,
		importeCobro: number = 0,
		fechaCobro: Date = null,
		codigoBarras: string = "",
		observacion: string = ""
	)
	{
		// this.idRecaudacionLote = idRecaudacionLote;
		// this.idReciboPublicacion = idReciboPublicacion;
		// this.idRegistroContableLote = idRegistroContableLote;
		// this.idPagoRendicionLote = idPagoRendicionLote;
		// this.idRecaudadora = idRecaudadora;
		this.numeroControl = numeroControl;
		this.numeroComprobante = numeroComprobante;
		this.codigoTipoTributo = codigoTipoTributo;
		this.numeroCuenta = numeroCuenta;
		this.codigoDelegacion = codigoDelegacion;
		this.numeroRecibo = numeroRecibo;
		this.importeCobro = importeCobro;
		this.fechaCobro = fechaCobro;
		this.codigoBarras = codigoBarras;
		// this.idUsuarioConciliacion = idUsuarioConciliacion;
		// this.fechaConciliacion = fechaConciliacion;
		this.observacion = observacion;
	}

	setFromObject = (row) =>
	{
		this.numeroControl = row.numeroControl ?? "";
		this.numeroComprobante = row.numeroComprobante ?? "";
		this.codigoTipoTributo = row.codigoTipoTributo ?? "";
		this.numeroCuenta = row.numeroCuenta ?? "";
		this.codigoDelegacion = row.codigoDelegacion ?? "";
		this.numeroRecibo = row.numeroRecibo ?? 0;
		this.importeCobro = row.importeCobro ?? 0;
		this.fechaCobro = row.fechaCobro ?? null;
		this.codigoBarras = row.codigoBarras ?? "";
		this.observacion = row.observacion ?? "";
	}

}

File: modules/tesoreria/models/recaudacion.py
from sqlalchemy import Column, BigInteger, String, Numeric, DateTime, ForeignKey
from sqlalchemy.orm import relationship

import sys, os
sys.path.insert(0, os.path.join(os.path.dirname(__file__), "..", ".."))
from shared.database import Base


class Recaudacion(Base):
    __tablename__ = "tesoreria_recaudaciones"

    id = Column(BigInteger, primary_key=True)
    id_recaudacion_lote = Column(BigInteger, ForeignKey("tesoreria_recaudacion_lotes.id"), nullable=False)
    numero_control = Column(String(50), nullable=True)
    numero_comprobante = Column(String(50), nullable=True)
    codigo_tipo_tributo = Column(String(50), nullable=True)
    numero_cuenta = Column(String(50), nullable=True)
    codigo_delegacion = Column(String(50), nullable=True)
    numero_recibo = Column(String(50), nullable=True)
    importe_cobro = Column(Numeric(18, 2), nullable=False)
    fecha_cobro = Column(DateTime, nullable=True)
    codigo_barras = Column(String(250), nullable=True)
    observacion = Column(String(500), nullable=True)

    lote = relationship("RecaudacionLote", back_populates="recaudaciones", lazy="joined")
